src/detext/run_detext.py

```diff
@@ -132,100 +136,117 @@ class Args(NamedTuple):
+    metadata_path: Optional[str] = None  # The metadata_path for converted avro2tf avro data.
 
     # tf-ranking related
     use_tfr_loss: bool = False  # whether to use tf-ranking loss.
     tfr_loss_fn: str = tfr.losses.RankingLossKey.SOFTMAX_LOSS  # tf-ranking loss
-    _tfr_loss_fn = {'choices': [tfr.losses.RankingLossKey.SOFTMAX_LOSS, tfr.losses.RankingLossKey.PAIRWISE_LOGISTIC_LOSS]}
-    tfr_lambda_weights: str = None  #
+    _tfr_loss_fn = {
+        'choices': [tfr.losses.RankingLossKey.SOFTMAX_LOSS, tfr.losses.RankingLossKey.PAIRWISE_LOGISTIC_LOSS]}
+    tfr_lambda_weights: Optional[str] = None  #
 
     use_horovod: bool = False  # whether to use horovod for sync distributed training
-
+    hvd_info: Optional[Dict[str, int]] = None
     # multitask training related
-    task_ids: List[int] = None  # All types of task IDs for multitask training. E.g., 1,2,3
+    task_ids: Optional[List[int]] = None  # All types of task IDs for multitask training. E.g., 1,2,3
     _task_ids = __comma_split_list(int)
-    task_weights: List[float] = None  # Weights for each task specified in task_ids. E.g., 0.5,0.3,0.2
+    task_weights: Optional[List[float]] = None  # Weights for each task specified in task_ids. E.g., 0.5,0.3,0.2
     _task_weights = __comma_split_list(float)
 
-
-def get_hparams():
+    def __late_init__(self):
+        hparams = self
+        # if epoch is set, overwrite training steps
+        if hparams.num_epochs is not None:
+            hparams = hparams._replace(num_train_steps=misc_utils.estimate_train_steps(
+                hparams.train_file,
+                hparams.num_epochs,
+                hparams.train_batch_size,
+                hparams.metadata_path is None))
+
+        # if num_eval_rounds is set, override steps_per_eval
+        if hparams.num_eval_rounds is not None:
+            hparams = hparams._replace(steps_per_eval=max(1, int(hparams.num_train_steps / hparams.num_eval_rounds)))
+
+        # For data sharding when using horovod
+        if hparams.use_horovod:
+            import horovod.tensorflow as hvd
+            hvd.init()
+            hparams = hparams._replace(
+                num_train_steps=hparams.num_train_steps // hvd.size(),
+                num_warmup_steps=hparams.num_warmup_steps // hvd.size(),
+                steps_per_eval=hparams.steps_per_eval // hvd.size(),
+                steps_per_stats=hparams.steps_per_stats // hvd.size(),
+                hvd_info={'rank': hvd.rank(), 'size': hvd.size()})
+
+        return hparams
+
+
+def get_hparams(argument: DetextArg = None):
     """
     Get hyper-parameters.
+    To be removed.
     """
-    hparams = tf.contrib.training.HParams(**Args()._asdict())
+    import warnings
+    warnings.warn("get_hparams is deprecated, call `run_detext` directly")
+    hparams = tf.contrib.training.HParams(**(argument or DetextArg.__from_argv__(error_on_unknown=False))._asdict())
```

Comment:
As `Hparams` is directly imported, can we replace this without the tf.xxx package prefix?

Reply:
This method is deprecated and not used. 
Mainly kept for backward compatibility.

Reply:
Misread the comment, but yeah, sure.